platforms/core-runtime/functional/src/main/java/org/gradle/internal/serialization/Cached.java

```diff
@@ -34,10 +34,15 @@ public static <T> Cached<T> of(Callable<T> computation) {
 
     public abstract T get();
 
+    public static boolean isResolving() {
```

Comment:
❓ How is this implemented for `provider { ... }`? Is there any possibility of unification?

Reply:
I think so, Mike shared some ideas, that is one of the things TBD.

Reply:
That is the bit I referred to here:

> changed platforms/core-runtime/functional/src/main/java/org/gradle/internal/serialization/Cached.java so the deprecation is not issued when computing the value (this may be revisited to integrate with EvaluationContext)

Reply:
That change is being done separately in https://github.com/gradle/gradle/pull/31164.  I will revert the ad-hoc scheme here.